### ISSUE TITLE
[Feauture/Help request] Skills using hologram

### ISSUE THREAD

**Issue by Lix3nn53** [Eniripsa96/SkillAPIKotlin#9]
I am making a trap skill which spawns a 3D trap model when used. And destroyed when an enemy steps on it. So I made a custom command to create and destroy the model. 

The problem is I can't find a way to remove the model when an entity other than player steps on it. (Since entities can't use commands)

Can you add something hologram/hologram destroy mechanic? If you think this request is too personal, can you suggest a way for me?

Btw here is my hologram:
`Location baseLocation = playerLocation;
                    ArmorStand model = (ArmorStand) baseLocation.getWorld().spawnEntity(baseLocation, EntityType.ARMOR_STAND);
                    ItemStack itemStack = new ItemStack(Material.IRON_PICKAXE);
                    ItemMeta itemMeta = itemStack.getItemMeta();
                    itemMeta.setCustomModelData(10000005);
                    itemMeta.setUnbreakable(true);
                    itemStack.setItemMeta(itemMeta);
                    model.setHelmet(itemStack);
                    model.setVisible(false);
                    model.setCustomName(ChatColor.DARK_GRAY + "< Trap " + ChatColor.DARK_GRAY + player.getName() + ChatColor.DARK_GRAY + " >");
                    model.setCustomNameVisible(true);
                    model.setInvulnerable(true);
                    model.setGravity(false);`

**Comment by Eniripsa96** [comment]
How are you detecting when the entity steps on it?

**Comment by Lix3nn53** [comment]
Haven't configure the skill yet but Location > Repeat > Area > (cancel repeat and remove model) should work?

**Comment by Eniripsa96** [comment]
How is the hologram spawning tied into the skill? Can a player have more than one active at a time?

I'm thinking one way you could do it for now is just track the trap by the player and run a command or [write a custom component](https://github.com/Eniripsa96/SkillAPI/wiki/_Tutorial-Coding-Components) that removes the trap.

Eventually, hologram support could be something I add, just won't be an initial feature so you could do the above to unblock yourself for now.

**Comment by Lix3nn53** [comment]
Right now when player uses skill it spawns a trap hologram with custom command. When an enemy player triggers a trap he also triggers a command to remove closest trap hologram.

But I am having a problem creating this skill even without models. Skill logic I think of:
increase value on use
Start repeat > condition value > area > decrease value when "target area" finds its first target

But this stops the repeat of last placed trap. Is there a way to make it stop repeat of the exact trap target triggered?

**Comment by Eniripsa96** [comment]
You should be able to just do a Cancel to stop the current repeat rather than using a condition to just skip parts.

**Comment by Lix3nn53** [comment]
Thanks didnt know it cancels repeat.
For holograms, I am trying to write a custom component as you mentioned.
Can I implement something like "hologram > hologramRemove" just like "repeat > cancel"?

**Comment by Eniripsa96** [comment]
Yeah, making components that way will end up generating a file that the editor will accept and let you build skills with it like any other component it has

**Comment by Lix3nn53** [comment]
I made a mechanic to spawn holograms: http://pasted.co/3d5a5178
But I don't know how to make the mechanic to remove it. (and make it work like repeat > cancel)
Can you guide me? thanks

**Comment by Lix3nn53** [comment]
Finally I am able to make this trap skill! Added two custom mechanics:
Hologram create: http://pasted.co/13c34aaa
Hologram destroy: http://pasted.co/823a23d0

I had to use "remember targets" mechanic on my hologram then select it by "remember" target and finally destroy it with Hologram destroy. 
Bear Trap.yml: http://pasted.co/e44fe6b2
